### PR TITLE
Bump static analysis levels

### PR DESCRIPTION
**Pull request by franmomu**
PHPstan to `max` and Psalm to `2`.

### CODE REVIEW

File: tests/App/AppKernel.php [hunk 1]
```diff
@@ -71,6 +71,11 @@ public function getProjectDir(): string
         return __DIR__;
     }
 
+    /**
+     * TODO: Use RoutingConfigurator when dropping support of symfony < 5.1.
```
Comment (jordisala1991) [inline tests/App/AppKernel.php:75]: see mediabundle master branch to see how it can be fixed

File: src/FieldDescription/FieldDescription.php [hunk 2]
```diff
@@ -65,19 +62,18 @@ public function describesCollectionValuedAssociation(): bool
...
     {
         $this->fieldMapping = $fieldMapping;
-
-        $this->mappingType = $this->mappingType ?: $fieldMapping['type'];
-        $this->fieldName = $this->fieldName ?: $fieldMapping['fieldName'];
+        $this->mappingType = $this->mappingType ?? $fieldMapping['type'];
     }
 
+    /**
+     * @psalm-suppress DocblockTypeContradiction
```
Comment (jordisala1991) [inline src/FieldDescription/FieldDescription.php:75]: can you add a note a about why it cant be fixed? like the others supress

File: src/FieldDescription/FilterTypeGuesser.php [hunk 3]
```diff
@@ -30,6 +30,9 @@
 
 final class FilterTypeGuesser implements TypeGuesserInterface
 {
+    /**
+     * @psalm-suppress DeprecatedConstant
```
Comment (jordisala1991) [inline src/FieldDescription/FilterTypeGuesser.php:34]: cant be avoided?

File: tests/FieldDescription/FilterTypeGuesserTest.php [hunk 4]
```diff
@@ -130,6 +130,8 @@ public function testGuessTypeNoAssociation(string $type, string $resultType, int
     }
 
     /**
+     * @psalm-suppress DeprecatedConstant
```
Comment (jordisala1991) [inline tests/FieldDescription/FilterTypeGuesserTest.php:133]: maybe here we can only test the non deprecated constants, so on next major we dont have to change it, and we avoid a supress

File: src/FieldDescription/FieldDescription.php (hunk 2 above)
Comment (VincentLanglet) [inline src/FieldDescription/FieldDescription.php:75]: @see https://github.com/vimeo/psalm/issues/5643 ?

File: src/FieldDescription/FilterTypeGuesser.php (hunk 3 above)
Comment (VincentLanglet) [inline src/FieldDescription/FilterTypeGuesser.php:34]: Maybe we should list the constant deprecated and when to remove them ?

File: src/FieldDescription/TypeGuesser.php [hunk 5]
```diff
@@ -22,6 +22,9 @@
 
 final class TypeGuesser implements TypeGuesserInterface
 {
+    /**
+     * @psalm-suppress DeprecatedConstant
```
Comment (VincentLanglet) [inline src/FieldDescription/TypeGuesser.php:26]: Maybe we should list the constant deprecated and when to remove them ?

File: tests/FieldDescription/FilterTypeGuesserTest.php (hunk 4 above)
Comment (VincentLanglet) [inline tests/FieldDescription/FilterTypeGuesserTest.php:133]: Maybe we should list the constant deprecated and when to remove them ?

File: tests/App/AppKernel.php (hunk 1 above)
Comment (franmomu) [reply tests/App/AppKernel.php:76]: 👍 I've removed the type declaration

File: src/FieldDescription/FieldDescription.php (hunk 2 above)
Comment (franmomu) [reply src/FieldDescription/FieldDescription.php]: Done with https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/602#discussion_r650489002

File: tests/FieldDescription/FilterTypeGuesserTest.php (hunk 4 above)
Comment (franmomu) [reply tests/FieldDescription/FilterTypeGuesserTest.php:133]: IMO it's good to test deprecated constants to see they work find and prevent to remove them accidentally, it's deprecated by `doctrine/mongodb-odm` so we do't know when we are going to remove them.

I would add these issues to `psalm.xml`, but when specifying the `DeprecatedConstant` issue, you have to say in which file and we could add more deprecated constants that have nothing to do with these ones and they would be suppressed too.

File: src/FieldDescription/FilterTypeGuesser.php (hunk 3 above)
Comment (franmomu) [reply src/FieldDescription/FilterTypeGuesser.php:34]: I've added comments to these constants

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by franmomu** [comment]
I'll create a pre-release

**Pull request merged**